### PR TITLE
Draft [models] feat: Add a modeling patch gen sample for qwen3

### PR DESCRIPTION
**Pull request by piyifan123**
### What does this PR do?

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`, `perf`, `ops`, `parallel`
  - If this PR involves multiple modules, separate them with `,` like `[ci, data, model]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][parallel, model] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: veomni/models/transformers/qwen3/patches/qwen3_gpu_patches.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        target="Qwen3MLP",
+        source_module="liger_kernel.transformers.swiglu",
+        source_name="LigerSwiGLUMLP",
+        description="Use LigerKernel SwiGLU MLP",
+    )
+)
+
+
+@config.replace_function("apply_rotary_pos_emb", description="Use LigerKernel rotary embedding")
+def apply_rotary_pos_emb_liger(
```
Comment (qqfish) [inline veomni/models/transformers/qwen3/patches/qwen3_gpu_patches.py:54]: Do you want to manage these in a common plance?

File: veomni/models/transformers/qwen3/patches/qwen3_gpu_patches.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        target="Qwen3RMSNorm",
+        source_module="liger_kernel.transformers.rms_norm",
+        source_name="LigerRMSNorm",
+        description="Use LigerKernel RMSNorm",
+    )
+)
+
+config.patches.append(
+    create_patch_from_external(
+        target="Qwen3MLP",
```
Comment (qqfish) [inline veomni/models/transformers/qwen3/patches/qwen3_gpu_patches.py:45]: Nit: The param name is a little bit confusing. source & target feels like changing from source to target.

File: veomni/models/transformers/qwen3/patches/qwen3_gpu_patches.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        source_module="liger_kernel.transformers.swiglu",
```
Comment (qqfish) [inline veomni/models/transformers/qwen3/patches/qwen3_gpu_patches.py]: Is it possible to actually import it instead of using a string?

File: veomni/models/transformers/qwen3/generated/patched_modeling_qwen3_gpu.py [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+            inputs_embeds=inputs_embeds,
+            use_cache=use_cache,
+            cache_position=cache_position,
+            **kwargs,
+        )
+
+        hidden_states = outputs.last_hidden_state
+        # Only compute necessary logits, and do not upcast them to float if we are not computing the loss
+        slice_indices = slice(-logits_to_keep, None) if isinstance(logits_to_keep, int) else logits_to_keep
+        logits = self.lm_head(hidden_states[:, slice_indices, :])
```
Comment (FoolPlayer) [inline veomni/models/transformers/qwen3/generated/patched_modeling_qwen3_gpu.py:504]: The logic to calculate loss here may changed if we use liger-kernel

File: veomni/models/transformers/qwen3/generated/patched_modeling_qwen3_gpu.py [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+            causal_mask_mapping = {
+                "full_attention": create_causal_mask(**mask_kwargs),
+            }
+            # The sliding window alternating layers are not always activated depending on the config
+            if self.has_sliding_layers:
+                causal_mask_mapping["sliding_attention"] = create_sliding_window_causal_mask(**mask_kwargs)
+
+        hidden_states = inputs_embeds
+        position_embeddings = self.rotary_emb(hidden_states, position_ids)
+
```
Comment (FoolPlayer) [inline veomni/models/transformers/qwen3/generated/patched_modeling_qwen3_gpu.py:417]: How to patch Qwen3Model.forward for SP case?

File: veomni/patchgen/README.md [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+```
+
+## Project Structure
+
+```
+veomni/patchgen/
+├── patch_spec.py              # Patch specification DSL
+├── codegen.py                 # AST-based code generator
+├── run_codegen.py             # CLI runner script
+├── patches/
```
Comment (FoolPlayer) [inline veomni/patchgen/README.md]: Can we still keep the position of the patch code and the generated code in the `model` dir

Comment (piyifan123) [reply veomni/patchgen/README.md]: SGTM moved.
